File: go/1089_Duplicate_Zeros.go
func duplicateZeros(arr []int)  {
    for i := 0; i < len(arr); i++ {
        if arr[i] == 0 { 
            for j := len(arr) - 2; j > i; j-- {
                arr[j + 1] = arr[j]
            }
            i++
            if i < len(arr) {
                arr[i] = 0
            }
        }
    }
}

File: php/1000/1089_Duplicate_Zeros.php
<?php

class Solution {

    /**
     * @param Integer[] $arr
     * @return NULL
     */
    function duplicateZeros(&$arr) {
        for ($i = 0; $i < count($arr) - 1; $i++) {
            if ($arr[$i] == 0) { 
                for ($j = count($arr) - 2; $j > $i; $j--) {
                    $arr[$j + 1] = $arr[$j];
                }
                $i++;
                $arr[$i] = 0;
            }
        }
    }
}

